spec/grape/endpoint_spec.rb

```diff
@@ -655,6 +655,38 @@ def app
     end
   end
 
+  describe '#declared; from a nested mounted endpoint - case 2' do
+    before do
+      mounted = Class.new(Grape::API) do
+        params { requires :inner_val, type: Integer }
+        get { declared(params) }
+      end
+
+      subject.format :json
+      subject.namespace :outer do
+        params { requires :outer_val, type: Integer }
```

Comment:
Yes.  You are correct.  I had mistakenly thought that once a params block was declared, it would remain in effect until superceded by another params block.  That turns out not to be the case; the block is consumed by the get in my example leaving nothing for the mounted endpoint to inherit.

Closing.